docs/topics/dokka-migration.md

```diff
@@ -77,157 +95,355 @@ After updating the Dokka Gradle plugin to v2, follow the migration steps applica
+
+Previous configuration:
+
+```kotlin
+tasks.dokkaHtmlMultiModule {
+    outputDirectory.set(rootDir.resolve("docs/api/0.x"))
+    includes.from(project.layout.projectDirectory.file("README.md"))
+}
+```
+
+New configuration:
+
+```kotlin
+dokka {
+    dokkaPublications.html {
+        outputDirectory.set(rootDir.resolve("docs/api/0.x"))
+        includes.from(project.layout.projectDirectory.file("README.md"))
+    }
+}
+```
+
+### Configure Dokka plugins
+
+Configuring built-in Dokka plugins with JSON has been deprecated in favor of a type-safe DSL. This change improves compatibility 
+with Gradle's incremental build system and improves task input tracking.
+
+Previous configuration:
+
+In DGP v1, Dokka plugins were configured manually using JSON. This approach caused issues with [registering task inputs](https://docs.gradle.org/current/userguide/incremental_build.html) 
+for Gradle up-to-date checks.
+
+Here is an example of the deprecated JSON-based configuration for the [Dokka Versioning plugin](https://kotl.in/dokka-versioning-plugin):
+
+```kotlin
+tasks.dokkaHtmlMultiModule {
+    pluginsMapConfiguration.set(
+        mapOf(
+            "org.jetbrains.dokka.versioning.VersioningPlugin" to """
+                { "version": "1.2", "olderVersionsDir": "$projectDir/dokka-docs" }
+                """.trimIndent()
+        )
+    )
+}
+```
+
+New configuration:
+
+DGP v2 replaces the JSON-based configuration with a type-safe DSL that is compatible with incremental builds.
+Use the `pluginsConfiguration{}` block to configure Dokka plugins in a type-safe way:
+
+```kotlin
+dokka {
+    pluginsConfiguration {
+        versioning {
+            version.set("1.2")
+            olderVersionsDir.set(projectDir.resolve("dokka-docs"))
         }
     }
-    ```
+}
+```
+
+### Configure custom Dokka plugins
+
+Dokka 2.0.0 allows you to extend its functionality by configuring custom plugins, 
+which enable additional processing or modifications to the documentation generation process.
+
+To configure a custom Dokka plugin, implement a custom `DokkaPluginParametersBaseSpec` 
+class in your `build.gradle.kts` file. The following example shows how to define and register a custom plugin parameter class:
+
+```kotlin
+// build.gradle.kts
+
+plugins {
+    id("org.jetbrains.dokka") version "2.0.0-Beta"
+}
+
+val dokkaScripts = layout.projectDirectory.dir("dokka-scripts")
+
+dokka {
+    pluginsConfiguration {
+        registerBinding(DokkaScriptsPluginParameters::class, DokkaScriptsPluginParameters::class)
+        register<DokkaScriptsPluginParameters>("DokkaScripts") {
+            scripts.from(dokkaScripts.asFileTree)
+        }
+    }
+}
+
+@OptIn(DokkaInternalApi::class)
+abstract class DokkaScriptsPluginParameters @Inject constructor(
+    name: String
+) : DokkaPluginParametersBaseSpec(name, "ca.solostudios.dokkascript.plugin.DokkaScriptsPlugin") {
```

Comment:
Who is `ca.solostudios`? I think we should avoid mentioning external namespaces.